src/API/REST/V3/Routes/Subscriptions/SubscriptionController.php

```diff
@@ -435,6 +456,41 @@ public function deleteItems($request): WP_REST_Response
         ], 200);
     }
 
+    /**
+     * Cancel a subscription.
+     *
+     * @unreleased
+     */
+    public function cancel_item($request)
+    {
+        $subscription = Subscription::find($request->get_param('id'));
+
+        if (! $subscription) {
+            return new WP_REST_Response(__('Subscription not found', 'give'), 404);
+        }
+
+        try {
+            if (give()->gateways->hasPaymentGateway($subscription->gatewayId)) {
+                $subscription->cancel(true);
+            } else {
+                $subscription->status = SubscriptionStatus::CANCELLED();
+                $subscription->save();
+            }
+
+            $trash = $request->get_param('trash');
+
+            if ($trash) {
+                $subscription->trash();
+            }
+
+            $response = $this->prepare_item_for_response($subscription->toArray(), $request);
```

Comment:
Let's use the SubscriptionViewModel class here to keep the same pattern from other Subscription endpoints. 

Example:

```
$item = (new SubscriptionViewModel($subscription))->includeSensitiveData(true)->exports();
$response = $this->prepare_item_for_response($item, $request);
```

Reply:
Resolved on 677bb556fd5f590904d4938f3d5d103675d9d895